### PR TITLE
Faster campaigns tests, part 1: the low-hanging fruit.

### PR DESCRIPTION
**Pull request by mrnugget**
This improves on the test times reported here https://github.com/sourcegraph/sourcegraph/issues/16320#issuecomment-738681917 by **3-4 seconds**.

Before:

```
go test ./enterprise/internal/campaigns -count=1
ok      github.com/sourcegraph/sourcegraph/enterprise/internal/campaigns        15.346s
```

After:

```
go test ./enterprise/internal/campaigns -count=1
ok      github.com/sourcegraph/sourcegraph/enterprise/internal/campaigns        12.438s
```

Speed bump is relatively stable.

---

What's in here:

* Rewriting one test to not use the database by introducing a tiny `FakeStore` that we could extend for other methods to inject a subset of the real `*Store`.
* **Big one**: instead of calling `dbtest.NewDB` (twice!) to create completely new test databases (as in: `CREATE DATABASE`-new) we only use `dbconn.Global`. That means we only need to create the "setup test database" cost (roughly ~3s) once when running tests.

### CODE REVIEW

File: enterprise/internal/campaigns/reconciler.go [hunk 1]
```diff
@@ -868,7 +868,11 @@ func loadExternalService(ctx context.Context, reposStore RepoStore, repo *repos.
 	return externalService, nil
 }
 
-func loadCampaign(ctx context.Context, tx *Store, id int64) (*campaigns.Campaign, error) {
+type getCampaigner interface {
+	GetCampaign(ctx context.Context, opts GetCampaignOpts) (*campaigns.Campaign, error)
+}
```
Comment (eseliger) [inline enterprise/internal/campaigns/reconciler.go:873]: Hmm 🤨 I get why we need it but it feels a bit like bloat

**Review by eseliger — APPROVED** [review]
LGTM. What was the biggest improvement coming from? Not setting up a new testing db?

File: enterprise/internal/campaigns/reconciler.go (hunk 1 above)
Comment (mrnugget) [reply enterprise/internal/campaigns/reconciler.go:873]: Agree, I'm not 100% convinced either, but I want to see whether this leads to anything or not. If it doesn't make sense, I'll remove it.

**Comment by mrnugget** [comment]
> LGTM. What was the biggest improvement coming from? Not setting up a new testing db?

Yep. The other test was just to see how much difference it makes (and it does) when running a single test with and without DB.

**Review by chrispine — COMMENTED** [review]
LGTM

**Pull request merged**